File: backend/Bioskopina.Services/Database/GenreBioskopina.cs
using System;
using System.Collections.Generic;

namespace Bioskopina.Services.Database;

public partial class GenreBioskopina
{
    public int Id { get; set; }

    public int GenreId { get; set; }

    public int MovieId { get; set; }

    public virtual Bioskopina Movies { get; set; } = null!;

    public virtual Genre Genre { get; set; } = null!;
}

File: bioskopina_mobile/lib/models/genre_bioskopina.dart
import 'package:json_annotation/json_annotation.dart';
import 'bioskopina.dart';
import 'genre.dart';

part 'genre_bioskopina.g.dart';

@JsonSerializable()
class GenreBioskopina {
  int? id;
  int? genreId;
  int? movieId;

  @JsonKey(name: 'movie')
  Bioskopina? movie;

  @JsonKey(name: 'genre')
  Genre? genre;

  GenreBioskopina(
      this.id,
      this.genreId,
      this.movieId,
      this.movie,
      this.genre,
      );

  factory GenreBioskopina.fromJson(Map<String, dynamic> json) =>
      _$GenreBioskopinaFromJson(json);

  Map<String, dynamic> toJson() => _$GenreBioskopinaToJson(this);
}
